src/preact/utils/xhr.js

```diff
@@ -0,0 +1,27 @@
+export const xhrUtils = {
+  /**
+   * Wrapper around `fetch` that returns the parsed JSON result
+   *
+   * @param {RequestInfo} url
+   * @param {RequestInit} [init]
+   * @return {Promise<*>}
+   */
+  fetchJson(url, init) {
+    return this.fetch(url, init).then((res) => res.json());
+  },
+
+  /**
+   * Wrapper around `fetch` that checks the status code
+   *
+   * @param {RequestInfo} url
+   * @param {RequestInit} [init]
+   * @return {Promise<Response>}
+   */
+  async fetch(url, init) {
+    const response = await self.fetch(url, init);
+    if (!response.ok) {
+      throw new Error(response.statusText);
+    }
+    return response;
+  },
+};
```

Comment:
How about if we export each function individually? I don't think grouping everything in one object adds any real benefit. I guess we'd want to call fetch something else...

```suggestion
  /**
   * Wrapper around `fetch` that returns the parsed JSON result
   *
   * @param {RequestInfo} url
   * @param {RequestInit} [init]
   * @return {Promise<*>}
   */
  export const fetchJson = (url, init) => {
    return this.fetch(url, init).then((res) => res.json());
  };

  /**
   * Wrapper around `fetch` that checks the status code
   *
   * @param {RequestInfo} url
   * @param {RequestInit} [init]
   * @return {Promise<Response>}
   */
export const fetch = async (url, init) => {
    const response = await self.fetch(url, init);
    if (!response.ok) {
      throw new Error(response.statusText);
    }
    return response;
  };
```

Reply:
Can we address this in https://github.com/ampproject/amphtml/issues/37465 ?  All these `utils` need some improvements.